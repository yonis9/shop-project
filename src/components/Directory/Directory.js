import React, { Component } from "react";
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { sectionsSelector } from '../../redux/directory/directory-selectors'

import MenuItem from '../MenuItem/MenuItem';

import './Directory.scss';

const  Directory = ({ sections }) => (
  <div className='directory-menu'>
       {sections.map(({ id, ...otherProps }) => (
          <MenuItem key={id} {...otherProps} />
      ))}
  </div>
)
        
    


const mapStateToProps = createStructuredSelector({
  sections: sectionsSelector
})

export default connect(mapStateToProps)(Directory);
