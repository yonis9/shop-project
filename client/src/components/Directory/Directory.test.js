import React from 'react';
import { shallow } from 'enzyme';

import { Directory } from './Directory';

it('should rendered Directory component', () => {
    expect(shallow(<Directory sections={[]} />)).toMatchSnapshot();
});