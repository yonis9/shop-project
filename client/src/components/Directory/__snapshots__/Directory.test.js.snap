// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered Directory component 1`] = `ShallowWrapper {}`;
